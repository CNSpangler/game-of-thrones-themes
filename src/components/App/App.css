  .fire {
    background-color: red;
  }

  .ice {
    background: blue;
  }

