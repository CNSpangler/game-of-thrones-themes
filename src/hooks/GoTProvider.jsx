import React, { createContext, useContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { fetchCharacters } from '../../services/GoT-API.js';

const GoTContext = createContext();

// wrapper to provide state to child components
export const GoTProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);
  const [page, setPage] = useState(6);

  useEffect(() => {
    fetchCharacters(page)
      .then(results => setCharacters(results));
  }, [page]);

  return (
    <GoTContext.Provider value={{ characters, page }}>
      {children}
    </GoTContext.Provider>
  );
};

GoTProvider.propTypes = {
  children: PropTypes.node
};

export const useCharacters = () => {
  const { characters } = useContext(GoTContext);
  return characters;
};

// export const useToggle = () => {
//   const { toggle } = useContext(GoTContext);
//   return toggle;
// };
