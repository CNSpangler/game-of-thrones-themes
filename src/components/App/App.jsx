import React from 'react';
import { GoTProvider } from '../../hooks/GoTProvider.jsx';
import Header from '../Header/Header.jsx';
import Characters from './Characters/Characters.jsx';

export default function App() {
  return (
    <GoTProvider>
      <Header />
      <Characters />
    </GoTProvider>
  );
}
