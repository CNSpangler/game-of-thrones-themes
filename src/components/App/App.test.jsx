import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { GoTProvider } from '../../hooks/GoTProvider';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(
      <GoTProvider>
        <App />
      </GoTProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
  
