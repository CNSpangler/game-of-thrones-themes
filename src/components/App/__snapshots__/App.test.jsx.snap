// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<ContextProvider
  value={
    Object {
      "characters": Array [],
      "fireIce": "fire",
      "page": 6,
      "toggle": [Function],
      "updatePage": [Function],
    }
  }
>
  <App />
</ContextProvider>
`;
